library/public/interactor.h

```diff
@@ -62,6 +71,50 @@ struct interaction_bind_t
         return this->inter;
     }
   }
+
+  /**
+   * Create and return an interaction bind from provided string
+   */
+  static interaction_bind_t parse(const std::string& str)
```

Comment:
should we make it inline and move the implementation details at the bottom of the file?